ql/math/interpolations/multicubicspline.hpp

```diff
@@ -474,14 +481,15 @@ namespace QuantLib {
         typedef typename c_splint::output_data output_data;
         typedef typename c_splint::dimensions dimensions;
         typedef typename c_splint::data data;
+        
         MultiCubicSpline(const SplineGrid& grid, const data_table &y,
-                         const std::vector<bool>& ae =
-                                             std::vector<bool>(20, false))
+                         const std::vector<bool>& ae = getDefaultAe())
```

Comment:
Apologies for the delay.

It's been decades since I looked into this code, but I'd try first to understand why the default is a vector of size 20 when `ae` should probably have a size dependent on the size of `grid`.  I suspect it's because the author thought that 20 was enough to cover most cases, but it's probably better to define the default as `const std::vector<bool>& ae = {}`; the constructor will recognize an empty vector as the default, and it will resize it to the proper size dependent on the grid and fill it with `false`.

This is likely to rule out a reference data member, but hopefully the interpolation is built once and used multiple times so the performance hit from the copy won't be that relevant.